File: archive/j/javascript/duplicate-character-counter.js
"use strict";

const USAGE = "Usage: please provide a string";

const isAlphaNum = (c) => /^[a-z0-9]$/i.test(c);

const countDuplicates = (str) => {
  const counts = new Map();

  for (const ch of str) {
    if (isAlphaNum(ch)) {
      counts.set(ch, (counts.get(ch) ?? 0) + 1);
    }
  }

  const duplicates = [];

  for (const [ch, count] of counts) {
    if (count > 1) {
      duplicates.push(`${ch}: ${count}`);
    }
  }

  return duplicates.length > 0
    ? duplicates.join("\n")
    : "No duplicate characters";
};

const run = () => {
  const [, , input] = process.argv;

  if (!input) {
    console.error(USAGE);
    process.exit(1);
  }

  console.log(countDuplicates(input));
};

run();

File: archive/j/java/DuplicateCharacterCounter.java
import java.util.*;

public class DuplicateCharacterCounter {
    public static void error() {
        System.out.println("Usage: please provide a string");
    }

    public static void main(String[] args) {
        if (args.length <= 0) {
            error();
        } else if (args[0].length() == 0) {
            error();
        } else {
            String inputStr = args[0];
            Map<Character, Integer> map = new HashMap<>();
            for (Character c : inputStr.toCharArray()) {
                map.put(c, map.getOrDefault(c, 0) + 1);
            }
            boolean flag = false;
            for (Character c : inputStr.toCharArray()) {
                if (map.get(c) > 1) {
                    flag = true;
                    System.out.printf("%c: %d\n", c, map.get(c));
                    map.put(c, 0);
                }
            }
            if (flag == false) {
                System.out.println("No duplicate characters");
            }
        }
    }
}